### ISSUE TITLE
Limiting files by type

### ISSUE THREAD

**Issue by AaronShea** [balderdashy/skipper#35]
I'm wondering what the best way to go about limiting the file types for uploads would be. 
For example, a function that can be called when the file is finished streaming, but before it is saved to disk/uploaded to s3. Allowing the upload to be stopped if the file is not the correct type.

**Comment by mikermcneil** [comment]
@aaron524 see https://github.com/balderdashy/skipper/issues/54#issuecomment-72048398

@johntom thanks for sharing!  I could use some help with https://github.com/balderdashy/skipper/commit/9916b4c7db31c7c66caeb7bd6552935c0f4dc3ac if you have some time to take a look (just released skipper 0.5.5 a couple of days ago btw)

**Comment by johntom** [comment]
Hi Mike,

Will take look and I'm on board for any help.

John

**Comment by plato-cambrian** [comment]
I'm doing something similar in #89, be warned (at least in my code) calling `req.file('file')` appears to crash the server if the `file` form value is empty/missing

**Comment by sailsbot** [comment]
Thanks for posting, @AaronShea.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
